spring-web/src/main/java/org/springframework/web/cors/CorsUtils.java

```diff
@@ -38,19 +39,22 @@ public abstract class CorsUtils {
 	 * Returns {@code true} if the request is a valid CORS one by checking {@code Origin}
 	 * header presence and ensuring that origins are different.
 	 */
-	public static boolean isCorsRequest(HttpServletRequest request) {
+	public static IsCorsRequestResult isCorsRequest(HttpServletRequest request) {
 		String origin = request.getHeader(HttpHeaders.ORIGIN);
 		if (origin == null) {
-			return false;
+			return IsCorsRequestResult.IS_NOT_CORS_REQUEST;
+		}
+		if (!(new UrlValidator().isValid(origin))) {
```

Comment:
Is this the right validation to use, given https://github.com/spring-projects/spring-framework/issues/33639 ?

Reply:
I wasn't aware of already existing validation mechanisms - will change ASAP